src/main/java/uk/gov/pay/connector/gateway/stripe/StripeNotificationService.java

```diff
@@ -263,21 +265,27 @@ private void processSourceNotification(StripeNotification notification) {
             ChargeEntity charge = maybeCharge.get();
 
             if (isChargeIn3DSRequiredOrReadyState(ChargeStatus.fromString(charge.getStatus()))) {
-                executePost3DSAuthorisation(charge, notification.getType());
+                executePost3DSAuthorisation(charge, notification.getType(), paymentIntent);
             }
 
         } catch (StripeParseException e) {
             logger.error("{} notification parsing for source object failed: {}", PAYMENT_GATEWAY_NAME, e);
         }
     }
 
-    private void executePost3DSAuthorisation(ChargeEntity charge, String notificationEventType) {
+    private void executePost3DSAuthorisation(ChargeEntity charge, String notificationEventType, StripePaymentIntent paymentIntent) {
         try {
             final StripeNotificationType type = byType(notificationEventType);
 
             Auth3dsResult auth3DsResult = new Auth3dsResult();
             auth3DsResult.setAuth3dsResult(getMappedAuth3dsResult(type));
 
+            Optional<StripeCharge> stripeCharge = paymentIntent.getCharge();
+            if (stripeCharge.isPresent() && stripeCharge.get().getPaymentMethodDetails() != null &&
+                    stripeCharge.get().getPaymentMethodDetails().getThreeDSecure() != null) {
+                    auth3DsResult.setThreeDsVersion(stripeCharge.get().getPaymentMethodDetails().getThreeDSecure().getVersion());
+            }
+
```

Comment:
Could use `ifPresent` so save having to call `get()` multiple times.

```suggestion
            paymentIntent.getCharge().ifPresent(stripeCharge -> {
                if (stripeCharge.getPaymentMethodDetails() != null &&
                        stripeCharge.getPaymentMethodDetails().getThreeDSecure() != null) {
                    auth3DsResult.setThreeDsVersion(stripeCharge.getPaymentMethodDetails().getThreeDSecure().getVersion());
                }
            });
```